react-components/src/architecture/base/README.md

```diff
@@ -0,0 +1,186 @@
+# Building and teesting
+
+Do the following commands:
+
+    yarn
+    yarn build
+    yarn storybook`
+
+This will automatically goto: https://localhost:3000/?modelUrl=primitivesOpen.
+
+Instead of yarn build, use can build with types by:
+
+    yarn tsc --noEmit
+
+When the StoryBook is open, goto `Architeture/Main`. The toolbar on the right hand side is the one I have made. Here some commands and tools are added.
+
+# Motivation
+
+The objectives with this framework it that is should be easy to use and extend in the future. I hope all new related functiality will be developed using this framework. More is decribed in this document: https://docs.google.com/presentation/d/1Y50PeqoCS5BdWyDqRNNazeISy1SYTcx4QDcjeMKIm78/edit?usp=sharing
+
+# Coding
+
+You will probably find the coding style somewhat different than usually done in Reveal. Some difference:
+
+- **Headers:** Headers are used to mark sections in a file. This is important for fast navigation and to keep the code in correct order. The order in the file is always:
+
+  - Public constants (exported)
+  - Instance fields.
+  - Instance properties
+  - Implementation of interface (if any)
+  - Overrides from the must basic base class (if any)
+  - :
+  - Overrides from the actually base class (if any)
+  - Instance methods (If many, break it up based on what they do)
+  - Static methods (if any)
+  - Local functions, types or classes used in this file only
+
+- **Virual methods:**: Since TypeScript doesn't allow the virtual keyword, all functions are virtual. This is a weekness for the application developer. I have tried to mark function as virtual by comments. You should not override other function than this. A virtual function normally brings in uncessesary complexibility and should general be used by care.
+
+  - Function with prefex `Core` are always virtual and protected. They can be overridden, but must call the same function for the base class. A function without the `Core` prefex is the one you should call from outside and is public. Experince shows that this pattern will make the code easier to maintain and extend in the future. For instance `remove/removeCore` and `initialize/initializeCore` in the `DomainObject`.
+  - Overridden methods should alway be marked by the override keyword. You will unfortunate not get any linting error when you forget this.
+  -
+
+- **Static methods and classes:** The linker don't like static method, epecially when the entire class have static method only. I wanted to use this more, but have used functions instead. I don't like this, because it doen't group similar function togeter. It will also be less readabble on the caller side, since you don't know what part of the code the function is coming from. In typescript we have for instance Math with act like a static class, but they use som magic to pretend it to be using interface.
```

Comment:
This can probably also be fixed by a lint rule

Reply:
Good, see slack message